waiter/src/waiter/token_watch.clj

```diff
@@ -0,0 +1,151 @@
+(ns waiter.token-watch
+  (:require [clojure.core.async :as async]
+            [clojure.data :as data]
+            [clojure.tools.logging :as log]
+            [metrics.counters :as counters]
+            [metrics.meters :as meters]
+            [metrics.timers :as timers]
+            [waiter.correlation-id :as cid]
+            [waiter.metrics :as metrics]
+            [waiter.token :as tkn]
+            [waiter.util.async-utils :as au]))
+
+(defn send-event-to-channels
+  "Given a list of watch channels and the event to send to each channel, send the event in a non blocking fashion and
+  close channels that error the async/put! operation. Returns a list of closed channels."
+  [watch-chans event]
+  (for [watch-chan watch-chans
+        :when
+        (try
+          (not (async/put! watch-chan event))
+          (catch AssertionError e
+            (log/error e "closing watch-chan due to error")
+            (async/close! watch-chan)
+            true))]
+    watch-chan))
```

Comment:
Returns the list of active chans, so ew do not have to run a remove/`disj` later:
```suggestion
  (->> watch-chans
    (map (fn send-event! [watch-chan]
           (try
              (when (async/put! watch-chan event)
                watch-chan)
              (catch AssertionError e
                (log/error e "closing watch-chan due to error")
                (async/close! watch-chan)
                nil))
    (remove nil?))  
```

Reply:
I went with reduce because of the side effects, as `map` and `remove` also returns a lazy sequence.